packages/react-strict-dom/src/native/index.js

```diff
@@ -10,12 +10,31 @@
 import typeof * as TStyleX from '@stylexjs/stylex';
 export * from '../types/StrictTypes';
 
+import * as React from 'react';
 import * as html from './html';
 import * as cssRaw from './stylex';
-import { ThemeProvider } from './modules/ThemeContext';
-import { typeof ThemeProvider as TThemeProvider } from './modules/ThemeContext';
+import { CustomPropertiesProvider } from './modules/ContextCustomProperties';
 
-const contexts = { ThemeProvider: ThemeProvider as TThemeProvider };
+type ProviderValue = $ReadOnly<{ [string]: string | number }>;
+
+type ProviderProps = $ReadOnly<{
+  children: React$MixedElement,
+  customProperties: ProviderValue
+}>;
+
+function ThemeProvider(props: ProviderProps): React$MixedElement {
+  const { children, customProperties } = props;
+
+  return customProperties ? (
+    <CustomPropertiesProvider children={children} value={customProperties} />
+  ) : (
+    children
+  );
+}
```

Comment:
This is just to preserve the legacy ThemeProvider export that is used internally in code written before we added StyleX theming APIs.